src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -22,6 +24,12 @@ const propTypes = {
 
     /** Beta features list */
     betas: PropTypes.arrayOf(PropTypes.string),
+
+    /** The policies which the user has access to */
+    policies: PropTypes.shape({
+        /** The policy name */
+        name: PropTypes.string,
+    }).isRequired,
```

Comment:
```suggestion
    policies: PropTypes.arrayOf(PropTypes.shape({
        /** The policy name */
        name: PropTypes.string,
    })).isRequired,
```

I think you have to wrap it in `PropTypes.arrayOf` because it is an array of object with that shape, right?

This is used later when calling `PolicyUtils.isMemberOfFreePolicy(props.policies)`, which access `policy.type`. Should we add the `type` property in the described `PropTypes.shape`?

Reply:
Interesting, I haven't seen this being used for the `policy` OnyxProp. A couple of examples:

```
yuwen@MemonbookProExpensify ~/Expensidev/App (yuwen-freePolicyDefaultRooms) App % git grep policies:                                  
src/components/ArchivedReportFooter.js:    policies: PropTypes.objectOf(PropTypes.shape({
src/components/ArchivedReportFooter.js:        policies: {
src/components/ReportWelcomeText.js:    policies: PropTypes.shape({
src/components/ReportWelcomeText.js:        policies: {
src/components/RoomNameInput.js:    policies: PropTypes.shape({
src/components/RoomNameInput.js:        policies: {
src/libs/Navigation/AppNavigator/MainDrawerNavigator.js:    policies: PropTypes.shape({
src/libs/Navigation/AppNavigator/MainDrawerNavigator.js:    policies: {
src/pages/ReportDetailsPage.js:    policies: PropTypes.shape({
src/pages/ReportDetailsPage.js:        policies: {
src/pages/ReportSettingsPage.js:    policies: PropTypes.shape({
src/pages/ReportSettingsPage.js:        policies: {
src/pages/RequestCallPage.js:    policies: PropTypes.shape({
src/pages/RequestCallPage.js:        policies: {
src/pages/home/HeaderView.js:    policies: PropTypes.shape({
src/pages/home/HeaderView.js:        policies: {
src/pages/home/ReportScreen.js:    policies: PropTypes.shape({
src/pages/home/ReportScreen.js:    policies: {
src/pages/home/report/ReportActionItemCreated.js:    policies: PropTypes.shape({
src/pages/home/report/ReportActionItemCreated.js:    policies: {},
src/pages/home/report/ReportActionItemCreated.js:    policies: {
src/pages/settings/InitialSettingsPage.js:    policies: PropTypes.objectOf(PropTypes.shape({
src/pages/settings/InitialSettingsPage.js:    policies: {},
src/pages/settings/InitialSettingsPage.js:        policies: {
src/pages/workspace/WorkspaceNewRoomPage.js:        policies: {
```

Reply:
Yeah I noticed that we had it like that in other places, and it made me think that maybe `PropTypes.shape` could be equivalent to `PropTypes.arrayOf(PropTypes.shape(...``, but I didn't see anything like that in the [documentation](https://reactjs.org/docs/typechecking-with-proptypes.html)

<img width="405" alt="image" src="https://user-images.githubusercontent.com/87341702/173877902-b7c0f9c6-56d0-403b-8eae-625adeb65052.png">

Maybe we have some bug and the check for PropTypes is not working perfectly well?

Anyway, you may consider this as a NAB, but would be good to have a look later and see if our type checks are working as expected

Reply:
I believe `objectOf` might be more appropriate:
If we look at the Policy action: 
https://github.com/Expensify/App/blob/0ceb6a43ebc87b0a2ea184859d837c36a3971a22/src/libs/actions/Policy.js#L18-L27

... there is a val/key pairing